packages/async-rewriter/src/async-writer-babel.spec.ts

```diff
@@ -2407,34 +2407,67 @@ function f() {
       });
     });
   });
-  describe('ClassDeclaration', () => {
-    const type = {
-      type: 'classdef',
-      returnType: {
-        type: 'Test',
-        attributes: {
-          regularFn: { type: 'function', returnType: signatures.Database, returnsPromise: false },
-          awaitFn: { type: 'function', returnType: signatures.unknown, returnsPromise: true }
-        }
-      }
-    };
+  describe('ThisExpression', () => {
     before(() => {
-      spy = sinon.spy(new SymbolTable([{ db: signatures.Database }, {}], signatures));
-      writer = new AsyncWriter(signatures, spy);
+      writer = new AsyncWriter(signatures);
+    });
+    it('errors outside of function', (done) => {
+      input = 'this.x';
+      try {
+        writer.compile(input);
+      } catch (e) {
+        expect(e.name).to.be.equal('MongoshUnimplementedError');
+        done();
+      }
     });
-    describe('adds methods to class', () => {
+    it('errors in regular function', (done) => {
+      input = 'function x() { this.x = 1 }';
```

Comment:
hmmm what if people instantiate classes like so:

```
function cat (name) {
  if (!(this instanceof Cat)) return new Cat(name)
  this._name = name
}
```
are we also saying `MongodbUnimplementedError` for something like the above?

Reply:
That function is on it's own right, not inside of a class declaration? Yeah it would be turned off since we can't do any inference on this at all. So the keyword would only be allowed within a method of a class. It's a bit strict but I think it's an edge case anyway because I don't anticipate people will be writing classes much with the shell anyway. If the error gets reported a lot via telemetry or feature requests we can try to revisit it

Reply:
it is on it's own, but that's how you'd write 'Classes' pre es6. I _feel_ like people are more likely to do this (it's quicker) than the actual es6 class declaration. But I could be wrong! This would be cool to monitor and see if lots of people hit this error!